packages/streams/logging.js

```diff
@@ -28,6 +28,56 @@ module.exports = {
   listLogFiles
 }
 
+class FileTimestampStreamWithDelete extends FileTimestampStream {
+  constructor(app, fullLogDir, filesToKeep, options){
+    super(options)
+    this.app = app
+    this.filesToKeep = filesToKeep
+    this.fullLogDir = fullLogDir
+    this.prevFilename = undefined
+  }
+
+  // This method of base class is called when new file name is contemplated
+  // So let's override it to check how many files are there and delete the oldest ones
+  newFilename() {
+    if (this.prevFilename !== this.currentFilename){ // Only do that after new file created
+      this.prevFilename = this.currentFilename
+      this.deleteOldFiles()
+    }
+    return super.newFilename()
+  }
+
+  deleteOldFiles(){
+    debug(`Checking for old log files`)
+    listLogFiles(this.app, (err, files) => {
+      if (err) {
+        console.error(err);
+      }else{
+        let numToDelete =  0;
+        if (files.length > this.filesToKeep) {
+          const sortedFiles = files.sort();
+          numToDelete =  files.length - this.filesToKeep;
+          debug(`Will delete ${numToDelete} files`)
+          for(let i = 0; i < numToDelete; i++){
+            const fileName = path.join(this.fullLogDir, sortedFiles[i])
+            debug(`Deleting ${fileName}`)
+            fs.unlink(fileName, (err) => {
+              if (err){
+                console.error(err)
+              }
+              else {
+                debug(`${fileName} was deleted`)
+              }
+            });
+          }
+        }else{
+          debug(`Found ${files.length}/${this.filesToKeep} log files, deleting ${Math.max(numToDelete, 0)}`)
```

Comment:
The log statement here is false - it is output only when nothing is deleted. I don't think we need the `else` branch for anything? And numToDelete can be `const`.

Reply:
Agree, it was useful while developing, but now just pollutes the outout.